### PR TITLE
Correct `transformer_size` parameter if needed for `ResponseSelector` during loading

### PR DESCRIPTION
**Pull request by jupyterjazz**
**Proposed changes**:
- Corrects `transformer_size` parameter in `ResponseSelector` which is None by default with a constant integer during model loading
- Otherwise it could lead to errors like #10519
- Credits to @mleimeister for an [informative and clear analysis](https://github.com/RasaHQ/rasa/issues/10519#issuecomment-999599561)

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/tensorflow/rasa_layers.py [hunk 1]
```diff
@@ -805,6 +806,7 @@ def _get_transformer_dimensions(
         transformer_units = config[TRANSFORMER_SIZE]
         if isinstance(transformer_units, dict):
             transformer_units = transformer_units[attribute]
+        transformer_units = transformer_units or DEFAULT_TRANSFORMER_SIZE
```
Comment (ancalita) [inline rasa/utils/tensorflow/rasa_layers.py:809]: Nitpicking: would it be more readable if this would be rewritten using the `if/else` ternary operator?

Comment (jupyterjazz) [reply rasa/utils/tensorflow/rasa_layers.py:809]: you mean changing to this?
`transformer_units = transformer_units if transformer_units else DEFAULT_TRANSFORMER_SIZE`

**Review by mleimeister — COMMENTED** [review]
Does the proposed fix in `RasaSequenceLayer` also take effect in the training phase of `ResponseSelector`, not just model loading? In that case the original function [ResponseSelector._warn_and_correct_transfomer_size](https://github.com/RasaHQ/rasa/blob/main/rasa/nlu/selectors/response_selector.py#L387) would be redundant and could be removed. 

However, if this gets problematic with the `ResponseSelector` tests (afaik they check that this function is run as expected), it could also be dealt with in a follow-up ticket.

**Comment by mleimeister** [comment]
For transparency, you could add to the docstring of `RasaSequenceLayer._get_transformer_dimensions` to clarify that `DEFAULT_TRANSFORMER_SIZE` is used if none or 0 given.

**Comment by mleimeister** [comment]
The existing [check](https://github.com/RasaHQ/rasa/blob/main/rasa/nlu/selectors/response_selector.py#L394) in `ResponseSelector._warn_and_correct_transfomer_size` tests for both `transformer_size == None` and `transformer_size < 1`, which allows to correct configs like this [test case](https://github.com/RasaHQ/rasa/blob/main/tests/nlu/selectors/test_selectors.py#L556), where `transformer_size = -1` is expected to be corrected to the default size. To be consistent the same check should be applied in the fix here.

File: rasa/utils/tensorflow/rasa_layers.py (hunk 1 above)
Comment (mleimeister) [reply rasa/utils/tensorflow/rasa_layers.py:809]: The existing [check](https://github.com/RasaHQ/rasa/blob/main/rasa/nlu/selectors/response_selector.py#L394) in `ResponseSelector._warn_and_correct_transfomer_size` tests for both `transformer_size is None` and `transformer_size < 1`, which allows to correct configs like this [test case](https://github.com/RasaHQ/rasa/blob/main/tests/nlu/selectors/test_selectors.py#L556), where `transformer_size = -1` is expected to be corrected to the default size. To be consistent the same check should be applied in the fix here.

**Comment by jupyterjazz** [comment]
> Does the proposed fix in `RasaSequenceLayer` also take effect in the training phase of `ResponseSelector`, not just model loading? In that case the original function [ResponseSelector._warn_and_correct_transfomer_size](https://github.com/RasaHQ/rasa/blob/main/rasa/nlu/selectors/response_selector.py#L387) would be redundant and could be removed.
> 
> However, if this gets problematic with the `ResponseSelector` tests (afaik they check that this function is run as expected), it could also be dealt with in a follow-up ticket.

Yes, it does and as you predicted removing the existing correction causes some problems in the tests. I'll create a separate ticket to address that. 
Just to clarify, should we remove the whole [function](https://github.com/RasaHQ/rasa/blob/85ffd7c87ff465430ad5387dd6c000e8ccf4bd5d/rasa/nlu/selectors/response_selector.py#L387) or should we leave the warning and disable the correction?

**Review by ancalita — APPROVED** [review]
LGTM 👍🏼

**Comment by mleimeister** [comment]
> Just to clarify, should we remove the whole [function](https://github.com/RasaHQ/rasa/blob/85ffd7c87ff465430ad5387dd6c000e8ccf4bd5d/rasa/nlu/selectors/response_selector.py#L387) or should we leave the warning and disable the correction?

Since the default is then set in `RasaSequenceLayer`, the warning could be moved there as well and the original function be removed completely. That way, the warning would be logged both when training and loading a model?

**Comment by mleimeister** [comment]
LGTM 🚀

**Review by mleimeister — APPROVED** [review]

**Pull request merged**